toolchain/mfc/test/test.py

```diff
@@ -215,25 +208,27 @@ def _handle_case(case: TestCase, devices: typing.Set[int]):
 
     cons.print(f"  [bold magenta]{case.get_uuid()}[/bold magenta]    {duration:6.2f}    {case.trace}")
 
-
 def handle_case(case: TestCase, devices: typing.Set[int]):
-    # pylint: disable=global-statement
-    global nFAIL
+    global nFAIL, nPASS, nSKIP
 
     nAttempts = 0
-
+    maxAttempts = ARG("max_attempts")
+    
+    if ARG("ci_mode"):
+        maxAttempts = 3
```

Comment:
See previous comment about `--ci-mode`.

Reply:
I added it because in the issue it said "It might also be useful to set the default to 0 retries (the default is 3) and only use 3 retries when in CI mode (this would require a new flag)."
Should I keep this or remove it? I have removed it for now in the lastest commit.

Reply:
Right, but as this currently is, there is a semantic gap between `--ci-mode` and `--max-retries`. It isn't obvious that these options would be coupled. For example, what should the behavior be if you pass both `--ci-mode` and `--max-retries 10` ?

The best is just to default `--max-retries` to 0 and let the CI runners explicitly pass `--max-retries` to 3 if need be. Actually, I believe this feature is no longer necessary. It was introduced to hide spurious failures in the macOS runner but I'm pretty sure the issue has gone away (or will in #599). You can keep for now if you don't want to do testing to be sure the original problem has gone away.

Reply:
Well, we can collectively blame @sbryngelson for this :)

Reply:
In engineering, always question the requirements - and authority.

Reply:
I removed it anyway. Everything should be up to date now.

Reply:
@henryleberre having 3 retries has actually saved us in CI _many_ times. It also comes up on Frontier (or used to), for mysterious reasons. @aricer123 I recommend keeping `--max-attempt=3` in CI.